### PR TITLE
Feature/order repository

### PR DESCRIPTION
**Pull request by C0Rs1K**

### CODE REVIEW

File: Shop.API/appsettings.json [hunk 1]
```diff
@@ -7,3 +7,4 @@
   },
   "AllowedHosts": "*"
 }
+0
```
Comment (Elizaveta564) [inline Shop.API/appsettings.json]: what is it?

File: Shop.DAL/Repositories/BaseRepository.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    public async Task<IEnumerable<T>> GetRange(Expression<Func<T, bool>> condition, CancellationToken cancellationToken)
+    {
+        return await _context.Set<T>().Where(condition).ToListAsync();
+    }
+
+    public async Task<T?> GetSingle(Expression<Func<T, bool>> condition, CancellationToken cancellationToken)
+    {
+        return await _context.Set<T>().Where(condition).SingleOrDefaultAsync();
```
Comment (Elizaveta564) [inline Shop.DAL/Repositories/BaseRepository.cs]: use FirstOrDefaultAsync()

File: Shop.DAL/Repositories/BaseRepository.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        _context.Set<T>().Remove(entity);
+    }
+
+    public async Task<IEnumerable<T>> GetRange(Expression<Func<T, bool>> condition, CancellationToken cancellationToken)
+    {
+        return await _context.Set<T>().Where(condition).ToListAsync();
+    }
+
+    public async Task<T?> GetSingle(Expression<Func<T, bool>> condition, CancellationToken cancellationToken)
```
Comment (Elizaveta564) [inline Shop.DAL/Repositories/BaseRepository.cs]: use AsNoTracking() for  GET methods

**Review by Elizaveta564 — APPROVED** [review]

File: Shop.API/appsettings.json (hunk 1 above)
Comment (C0Rs1K) [reply Shop.API/appsettings.json]: It's a prank there's a camera over there🥲

Comment (C0Rs1K) [reply Shop.API/appsettings.json]: Solved
My bad

File: Shop.DAL/Repositories/BaseRepository.cs (hunk 3 above)
Comment (C0Rs1K) [reply Shop.DAL/Repositories/BaseRepository.cs]: Solved in base repository task

File: Shop.DAL/Repositories/BaseRepository.cs (hunk 2 above)
Comment (C0Rs1K) [reply Shop.DAL/Repositories/BaseRepository.cs]: Solved in base repository task

**Pull request merged**